internal/routing/reader.go

```diff
@@ -88,6 +88,56 @@ func (r *routing) LocalSubnet() (defaultSubnet net.IPNet, err error) {
 	return defaultSubnet, fmt.Errorf("cannot find default subnet in %d routes", len(routes))
 }
 
+func (r *routing) LocalSubnets() (localSubnets []net.IPNet, err error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return localSubnets, fmt.Errorf("cannot find local subnet: %w", err)
+	}
+
+	localLinks := make(map[int]bool)
```

Comment:
To make a set in Go, have the key as your set value and the value as the empty struct (=0 byte).

That way it's also clearer what the intent is and that the map value don't carry any information 😉 

```suggestion
	localLinks := make(map[int]struct{})
```